e2e/serial/dappInteractions/5_maliciousDapp.test.ts

```diff
@@ -0,0 +1,125 @@
+import 'chromedriver';
+import 'geckodriver';
+import { WebDriver } from 'selenium-webdriver';
+import {
+  afterAll,
+  afterEach,
+  beforeAll,
+  beforeEach,
+  describe,
+  expect,
+  it,
+} from 'vitest';
+
+import {
+  checkWalletName,
+  delayTime,
+  findElementByTestId,
+  findElementByTestIdAndClick,
+  findElementByText,
+  findElementByXPath,
+  getAllWindowHandles,
+  getExtensionIdByName,
+  getRootUrl,
+  getWindowHandle,
+  goToPopup,
+  importWalletFlow,
+  initDriverWithOptions,
+  takeScreenshotOnFailure,
+  waitAndClick,
+} from '../../helpers';
+import { TEST_VARIABLES } from '../../walletVariables';
+
+let rootURL = getRootUrl();
+let driver: WebDriver;
+
+const browser = process.env.BROWSER || 'chrome';
+const os = process.env.OS || 'mac';
+
+describe('App interactions flow', () => {
+  beforeAll(async () => {
+    driver = await initDriverWithOptions({
+      browser,
+      os,
+    });
+    const extensionId = await getExtensionIdByName(driver, 'Rainbow');
+    if (!extensionId) throw new Error('Extension not found');
+    rootURL += extensionId;
+  });
+
+  beforeEach<{ driver: WebDriver }>(async (context) => {
+    context.driver = driver;
+  });
+
+  afterEach<{ driver: WebDriver }>(async (context) => {
+    await takeScreenshotOnFailure(context);
+  });
+
+  afterAll(() => driver.quit());
+
+  it('should be able import a wallet via seed', async () => {
+    await importWalletFlow(driver, rootURL, TEST_VARIABLES.EMPTY_WALLET.SECRET);
+  });
+
+  it('should display account name', async () => {
+    await checkWalletName(driver, rootURL, TEST_VARIABLES.EMPTY_WALLET.ADDRESS);
+  });
+
+  it('should be able to go to setings', async () => {
+    await goToPopup(driver, rootURL);
+    await findElementByTestIdAndClick({ id: 'home-page-header-right', driver });
+    await findElementByTestIdAndClick({ id: 'settings-link', driver });
+  });
+
+  it('should be able to connect to hardhat', async () => {
+    await findElementByTestIdAndClick({ id: 'connect-to-hardhat', driver });
+    const button = await findElementByText(driver, 'Disconnect from Hardhat');
+    expect(button).toBeTruthy();
+    await findElementByTestIdAndClick({
+      id: 'navbar-button-with-back',
+      driver,
+    });
+  });
+
+  it('should be able to navigate to the malicious app and click connect', async () => {
+    await delayTime('long');
+    await driver.get('https://test-dap-welps.vercel.app/');
+    const dappHandler = await getWindowHandle({ driver });
+
+    const button = await findElementByTestId({
+      id: 'rk-connect-button',
+      driver,
+    });
+    expect(button).toBeTruthy();
+    await waitAndClick(button, driver);
+    await delayTime('long');
+
+    // todo: clean this xpath up
+    const rainbowButton = await findElementByXPath(
+      driver,
+      '/html/body/div[2]/div/div/div[2]/div/div/div/div/div[1]/div[2]/div[2]/div[1]/button/div/div',
```

Comment:
shy xpath and not testid?

Reply:
this is on wayne's test dapp and the test ID wasn't working for me 😅 Let me try again tho bc it's been a few weeks. I believe it was the RK test ID: `rk-wallet-option-rainbow`